.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+# Build the package and run tests, using the latest trifinger_user Apptainer
+# image.
+# The Apptainer image has a full ROBOT_FINGERS workspace installed, so all
+# dependencies are there and only the checked package needs to be build, not the
+# whole workspace.
+#
+# Note: The Apptainer image only gets automatically rebuild, if something in the
+# image definition changes, not when other packages are changed.  So to avoid
+# that the workspace in the container gets outdated, manual builds need to be
+# triggered regularly.
+
+name: Build and Test
+
+on:
+  pull_request:
+  # NOTE: By also running on master (=the default branch), we have a cache for
```

Comment:
Note sure what this implies. Is this a warning ? (i.e. something weird will happen because the cache of the master branch will be used ?)

Reply:
No, it is desired in this case.  If I understood it correctly, it works as follows:
The cache of branch A is in general not visible for branch B, so B would create its own cache.  However, caches of the default branch (i.e. master) are visible to other branches.  So if there is a cache for master, other branches don't need to create their own cache but can use the one of the master branch.
Since the Apptainer image that is used is the same on all branches, having a common cache seems useful to me here, as it reduces the number of unnecessary downloads.

Note that in practise this is only really relevant if there are multiple pull requests per week, as the cache anyway gets renewed every week.  So not sure how useful it really is in our case but in general it seems to be the proper way of doing it to me.

Reply:
I rephrased the comment a bit to hopefully make it clearer.